core/src/HIP/Kokkos_HIP_KernelLaunch.hpp

```diff
@@ -165,6 +165,7 @@ struct HIPParallelLaunch<
                                        MinBlocksPerSM>
           <<<grid, block, shmem, hip_instance->m_stream>>>(d_driver);
 
+      HIP_SAFE_CALL(hipGetLastError());
```

Comment:
I guess we could do
```suggestion
      HIP_SAFE_CALL(hipPeekAtLastError());
```
instead as suggested in https://stackoverflow.com/questions/14038589/what-is-the-canonical-way-to-check-for-errors-using-the-cuda-runtime-api/14038590.

Reply:
Good point, I didn't check it -- I had added to check for errors launching the kernel itself due to overuse of LDS.

Reply:
Let me run on one of the tree systems

Reply:
Still seeing the segfault as well, checking w/ a debug build.

Reply:
Ok -- as identified in the Confluence page, there is a separate issue in HIP causing the segfault.  I still think we probably want to check the error status of the kernel launch to avoid issues like this.

I don't think get vs peek maters too much.  If we want to check the error status of the kernel run (rather than the launch) we can wrap the fence call in a `HIP_SAFE_CALL`

Reply:
We already do that. :slightly_smiling_face:

Reply:
Which part? I think the fence is still not protected, no?

https://github.com/kokkos/kokkos/blob/develop/core/src/HIP/Kokkos_HIP_Instance.cpp#L165

Reply:
Oh, you're right. I was looking for the global fence that uses https://github.com/kokkos/kokkos/blob/2bc24d65bacab41f184ab159c94364eec19e42e8/core/src/HIP/Kokkos_HIP_Space.cpp#L748.
Yes, it probably makes sense to do that there as well. But let's discuss that in another pull request.

Reply:
See https://github.com/kokkos/kokkos/pull/3245.